### PR TITLE
[fix] torch.repeat : dim-0 backward

### LINKED ISSUE #45201: Backward of repeat() crash if repeated size is 0
Repro:

```python
import torch

a = torch.rand(10, requires_grad=True)

b = a.repeat(0)

b.sum().backward()
```

Fails with:
```
Traceback (most recent call last):
  File "foo.py", line 9, in <module>
    b.sum().backward()
  File "/usr/local/lib/python3.7/site-packages/torch/tensor.py", line 185, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/usr/local/lib/python3.7/site-packages/torch/autograd/__init__.py", line 127, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: chunk expects `chunks` to be greater than 0, got: 0
```

This should return a gradient full of 0s

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #45201

### CODE REVIEW

**Comment by kshitij12345** [comment]
@albanD Please review :)

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -705,7 +705,11 @@ Tensor sum_tensorlist(TensorList tl) {
   return sum;
 }
 
-Tensor repeat_backward(Tensor grad, int64_t input_dims, IntArrayRef repeats) {
+Tensor repeat_backward(Tensor grad, int64_t input_dims, IntArrayRef repeats, IntArrayRef input_shape) {
```
Comment (albanD) [inline torch/csrc/autograd/FunctionsManual.cpp:708]: nit: you can remove the `input_dims` argument now that you have all the shapes.

File: test/test_autograd.py [hunk 2]
```diff
@@ -4459,6 +4459,14 @@ def test_power_function(self):
         c.backward()
         self.assertEqual(b.grad, torch.tensor([-inf, 0., 0.]))
 
+    def test_repeat_zero_dim(self):
```
Comment (albanD) [inline test/test_autograd.py:4462]: You don't have to do a custom test here. You can simply add a new entry [here](https://github.com/pytorch/pytorch/blob/da4033d32adafd6fe0fa3a3727fcb2b55c19e2e9/torch/testing/_internal/common_methods_invocations.py#L881) that specifies the input size and the arguments.
So something like `('repeat', (2), (0,), 'zero_repeat'),` will do the same thing.

**Comment by albanD** [comment]
Thanks for sending a PR!
I added a few comments inline above.

**Comment by kshitij12345** [comment]
@albanD Have addressed the comments. PTAL:)

**Review by albanD — APPROVED** [review]
Perfect! Thanks for the update!

**Pull request closed without merging**